metadata/src/main/java/org/apache/kafka/metadata/BrokerRegistration.java

```diff
@@ -50,6 +54,7 @@ public static class Builder {
         private boolean fenced = false;
         private boolean inControlledShutdown = false;
         private boolean isMigratingZkBroker = false;
+        private Set<Uuid> directories;
```

Comment:
I think a sorted array of directories would be easier to work with.

We need to pass an array to most of places that are using this.

Also, if the array is sorted, we can always do a binary search for whatever value we want (although given the list is short, that might not matter much either).